pkg/engine/internal/scheduler/scheduler.go

```diff
@@ -34,9 +41,7 @@ type Scheduler struct {
 	initOnce sync.Once
 	svc      services.Service
 
-	// local is the local listener used for communication with workers running
-	// within the same process. May be unused if no workers are running locally.
-	local *wire.Local
+	workerListener wire.Listener // used for local and remote transport
```

Comment:
nit: should we rename this as `listener`? worker prefix is a bit confusing to me